linkerd/main/src/main/scala/io/buoyant/Linkerd.scala

```diff
@@ -76,4 +80,26 @@ object Linkerd extends App {
 
     Linker.parse(configText)
   }
+
+  /**
+   * Trap termination signals and triggers an App.close for a graceful shutdown.
+   * Shutdown hook is not used because it has, at least, the following problems:
+   * <ul>
+   *   <li>LogManager uses a shutdown hook which makes nothing to be logged during shutdown
+   *   <li>TracerCache uses a shutdown hook to flush
+   * </ul>
+   */
+  private def registerTerminationSignalHandler(shutdownGraceMs: Int): Unit = {
+    val shutdownHandler = new SignalHandler {
+      override def handle(sig: Signal): Unit = {
+        log.info("Received %s. Shutting down ...", sig)
+        Await.result(close(Duration.fromMilliseconds(shutdownGraceMs)))
+      }
+    }
+
+    Signal.handle(new Signal("INT"), shutdownHandler)
+    Signal.handle(new Signal("TERM"), shutdownHandler)
+    ()
```

Comment:
If I don't add that, sbt linkerd/assembly complains about:
```
[error] /Users/diego/Repo/linkerd-medallia/linkerd/main/src/main/scala/io/buoyant/Linkerd.scala:104: discarded non-Unit value
[error]     Signal.handle(new Signal("TERM"), shutdownHandler)
[error]                  ^
[error] one error found
```

One alternative is to add a val _ = ...